procedures.sql

```diff
@@ -0,0 +1,1136 @@
+END;
+GO
+-- EXEC AddStudentToMeeting 17, 23, '2021-01-01 12:00:00'
+
+CREATE PROCEDURE RemoveStudentFromMeeting
+    @student_id INT,
+    @meeting_id INT
+AS
+BEGIN
+    DELETE FROM StudentMeeting
+    WHERE student_id = @student_id AND meeting_id = @meeting_id;
+END;
+GO
+-- EXEC RemoveStudentFromMeeting 17, 23
+
+CREATE PROCEDURE PayForMeeting
+    @student_id INT,
+    @meeting_id INT,
+    @payment_date DATETIME = CURRENT_TIMESTAMP
+AS
+BEGIN
+    UPDATE StudentMeeting
+    SET payment_date = @payment_date
+    WHERE student_id = @student_id AND meeting_id = @meeting_id;
+END;
+GO
+-- EXEC PayForMeeting 17, 23
+
+
+
+CREATE PROCEDURE AddStudentToCourse
+    @student_id INT,
+    @course_id INT,
+    @advance_payment_date DATETIME = NULL,
+    @full_payment_date DATETIME = NULL
+AS
+BEGIN
+    INSERT INTO StudentCourse (student_id, course_id, advance_payment_date, full_payment_date)
+    VALUES (@student_id, @course_id, @advance_payment_date, @full_payment_date);
+END;
+GO
+-- EXEC AddStudentToCourse 17, 23, @advance_payment_date = '2021-01-01 12:00:00'
+
+CREATE PROCEDURE RemoveStudentFromCourse
+    @student_id INT,
+    @course_id INT
+AS
+BEGIN
+    DELETE FROM StudentCourse
+    WHERE student_id = @student_id AND course_id = @course_id;
+END;
+GO
+-- EXEC RemoveStudentFromCourse 17, 23
+
+CREATE PROCEDURE PayForCourse
+    @student_id INT,
+    @course_id INT,
+    @advance_payment_date DATETIME = CURRENT_TIMESTAMP,
+    @full_payment_date DATETIME = CURRENT_TIMESTAMP
+AS
+BEGIN
+    UPDATE StudentCourse
+    SET full_payment_date = @full_payment_date,
+        advance_payment_date = @advance_payment_date
+    WHERE student_id = @student_id AND course_id = @course_id;
+END;
+GO
+-- EXEC PayForCourse 17, 23
+
+
+
+CREATE PROCEDURE AddStudentToStudies
+    @student_id INT,
+    @studies_id INT,
+    @registration_payment_date DATETIME = NULL
+AS
+BEGIN
+    INSERT INTO StudentStudies (student_id, studies_id, registration_payment_date)
+    VALUES (@student_id, @studies_id, @registration_payment_date);
+END;
+GO
+-- EXEC AddStudentToStudies 17, 23, @registration_payment_date = '2021-01-01 12:00:00'
+
+CREATE PROCEDURE RemoveStudentFromStudies
+    @student_id INT,
+    @studies_id INT
+AS
+BEGIN
+    DELETE FROM StudentStudies
+    WHERE student_id = @student_id AND studies_id = @studies_id;
+
+    DELETE StudentSemester
+    FROM StudentSemester
+             INNER JOIN Semester ON StudentSemester.semester_id = Semester.id AND Semester.studies_id = @studies_id
+    WHERE student_id = @student_id;
+END;
+GO
+-- EXEC RemoveStudentFromStudies 17, 23
+
+CREATE PROCEDURE PayForStudies
+    @student_id INT,
+    @studies_id INT,
+    @registration_payment_date DATETIME = CURRENT_TIMESTAMP
+AS
+BEGIN
+    UPDATE StudentStudies
+    SET registration_payment_date = @registration_payment_date
+    WHERE student_id = @student_id AND studies_id = @studies_id;
+END;
+GO
+-- EXEC PayForStudies 17, 23
+
+CREATE PROCEDURE SendGraduationCertificate
```

Comment:
I have no idea how to send a letter with SQL. You meant `register_sending`, didn't you?

Reply:
as our whole system is just a SQL database, sending a letter is equivalent with saving this fact to a database, but assuming it really bother you, I may change the name

Reply:
`sending a letter is equivalent with saving this fact to a database` - it's an assumption. I'd be even willing to agree with you if you be consistent and apply it everywhere, but you don't